opencog/query/PatternUtils.cc

```diff
@@ -117,21 +116,18 @@ void get_connected_components(
 			size_t nc = components.size();
 			for (size_t i = 0; i<nc; i++)
 			{
-				std::set<Handle> cur_vars(component_vars[i]);
```

Comment:
add const whereever you want. however, in the atomspace, realize this:  under the covers, each atom has a use_count in it, which is incremented whenever a a Handle or AtomPtr points at it, and is decremented when a Handle or AtomPtr is destroyed.  So Atoms are kind-of-like const, but not really.  No clue how the compiler handles this.

Next: Handles are almost const, but not quite: if a Handle only has a uuid in it, but doesn't have a pointer to the actual atom, that needs to be fixed up.  It all happens deep in the guts, and so it is kind-of-like almost const, but I imagine that there could be some really weird bugs if the compiler is fooled. I don't know. So far, no problem.

The only place where I would expect this would be in the postgress side, where the database only stores uuid's and pointers are created later.  There's also a few oddball cases in the scheme bindngs which use only uuids ...